src/frontend/src/optimizer/plan_node/predicate_pushdown.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use paste::paste;
+
+use super::*;
+use crate::utils::Condition;
+use crate::{for_batch_plan_nodes, for_stream_plan_nodes};
+/// The trait for predicate pushdown, only logical plan node will use it, though all plan node impl
+/// it.
+pub trait PredicatePushdown {
+    /// Push predicate down for every logical plan node.
+    ///
+    /// Sometimes, you will find it hard or just want to delay the implementation of predicate
+    /// pushdown, then you can use [`gen_filter_and_pushdown`] to generate a `LogicalFilter` above
+    /// current node and do predicate pushdown for its input.
+    ///
+    /// Also, if current logcial plan node doesn't have any input(e.g. `LogicalScan`), you can use
+    /// [`gen_filter`] to generate a `LogicalFilter` above it.
+    ///
+    /// Please note that `LogicalFilter::create` will NOT create a `LogicalFilter` if `predicate`
+    /// is always true, so feel free to use these two helper functions.
```

Comment:
Still feel verbose. There's no need to leave such pedagogical guides. Just describe behavior. e.g. the three different kinds of operators as you mentioned in RFC